src/index.js

```diff
@@ -1,3 +1,5 @@
+const visited = Symbol('visited');
```

Comment:
Instead of mutating the path, this should use a `WeakMap`.

Reply:
Good call, updated to use this